site/it/community/contribute/code_style.md

```diff
@@ -0,0 +1,164 @@
+```
+
+### Versioni di Python supportate
+
+TensorFlow supporta Python 2.7 e Python >= 3.4. Vedere la
+[guida di installazione](https://www.tensorflow.org/install) per i dettagli.
+
+Vedere lo 
+[stato del build continuo](https://github.com/tensorflow/tensorflow/blob/master/README.md#continuous-build-status)
+di TensorFlow per i build ufficiali e quelli supportati dalla comunità.
+
+#### Compatibilità con Legacy Python
+
+TensorFlow supporterà Legacy Python (Python 2.7) fino al 
+[1 Gennaio, 2020](https://groups.google.com/a/tensorflow.org/forum/#!searchin/announce/python$202.7%7Csort:date/announce/gVwS5RC8mds/dCt1ka2XAAAJ).
+Fino ad allora, tutto il codice bisognerà che sia compatibile con le versioni di Python elencate sopra.
+
+Queste linee dovranno essere presenti in ogni file Python:
+
+
+```python
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+```
+
+Usate `six` per scrivere codice compatibile (per esempio, `six.moves.range`).
+
+
+## Style di codifica C++
+
+Le modifiche al codice C++ di TensorFlow dovrebbero essere conformi alla 
+[Guida Google di Stile C++](https://google.github.io/styleguide/cppguide.html). 
+Usare `clang-tidy` per controllare le vostre modifiche al codice C/C++.
+
+Per installare  `clang-tidy` su Ubuntu 16+:
+
+
+```bash
+$ apt-get install -y clang-tidy
+```
+
+Potete controllare un file C/C++ con il seguente:
+
+```bash
+$ clang-format <my_cc_file> --style=google > /tmp/my_cc_file.cc
+$ diff <my_cc_file> /tmp/my_cc_file.cc
+```
+
+## Altri linguaggi
+
+*   [Guida di Stile Google per Java](https://google.github.io/styleguide/javaguide.html)
+*   [Guida di Stile Google per JavaScript](https://google.github.io/styleguide/jsguide.html)
+*   [Guida di Stile Google per Shell](https://google.github.io/styleguide/shell.xml)
+*   [Guida di Stile Google per Objective-C](https://google.github.io/styleguide/objcguide.html)
+
+
+
+
+## Convenzioni ed usi speciali in TensorFlow
+
+### Tensori
+
+*   Le operazioni che hanno a che trattano batch possono ipotizzare che la **prima dimensione** di
+    un Tensore sia la dimensione del batch.
+*   In molti modelli, l'**ultima dimensione** è il numero di _canali_.
+*   Le dsimensioni escluse la prima e l'ultima, di solito, rappresentano le dimensioni dell _spazio_
+    : la lunghezza della sequenza, o la dimensione dell'immagine.
+    *   Quando possibile, usate un operatore sovraccarivcato tra Tensori, invece di una funzione di TensorFlow. 
+    Per esempio, noi preferiamo `**`, `+`, `/`, `*`, `-`, `and/or` a
+    `tf.pow`, `tf.add`, `tf.divide`, `tf.multiply`, `tf.subtract`, e `tf.logical_*` —
+    a meno che sia richiesto uno specifico nome per l'operazione.
+
+
+### Operazioni Python
+
+Un' _operazione Python_ è una funzione che, dati in input tensori e parametri,
+crea una parte del grafo, e ritorna come risultato dei tensori.
+*   Il primo argomento dovrà essere un tensore, seguito dai parametri Python di base.
```

Comment:
Ok, I might be answering myself if the next bullet point is explaining the tensors that are in the first argument. If so I would translate it more like "Il primo argomento dovrebbe essere costituto da tensori", using a plural form instead of a singular one. I'll need a confirmation for validating my thoughts